protocol/ip.go

```diff
@@ -44,7 +44,9 @@ func NewIPv4() *IPv4 {
 }
 
 func (i *IPv4) Len() (n uint16) {
-	i.IHL = 5
+	if i.IHL < 5 {
```

Comment:
Maybe just remove i.IHL = 5, and check the IHL in `func (i *IPv4) UnmarshalBinary(data []byte) error {` ?